generator/generator.go

```diff
@@ -1167,95 +1167,136 @@ func (g *generator) writeBoolExpressionAttribute(indentLevel int, attr parser.Bo
+	if _, err = g.w.Write("\n"); err != nil {
+		return err
+	}
+	if _, err = g.w.WriteIndent(indentLevel, "_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString("+vn+".Call)\n"); err != nil {
+		return err
+	}
+	return g.writeErrorHandler(indentLevel)
+}
+
+func (g *generator) writeExpressionAttributeValueDefault(indentLevel int, attr parser.ExpressionAttribute) (err error) {
+	var r parser.Range
+	vn := g.createVariableName()
+	// var vn string
+	if _, err = g.w.WriteIndent(indentLevel, "var "+vn+" string\n"); err != nil {
+		return err
+	}
+	// vn, templ_7745c5c3_Err = templ.JoinStringErrs(
+	if _, err = g.w.WriteIndent(indentLevel, vn+", templ_7745c5c3_Err = templ.JoinStringErrs("); err != nil {
+		return err
+	}
+	// p.Name()
+	if r, err = g.w.Write(attr.Expression.Value); err != nil {
+		return err
+	}
+	g.sourceMap.Add(attr.Expression, r)
+	// )
+	if _, err = g.w.Write(")\n"); err != nil {
+		return err
+	}
+	// Attribute expression error handler.
+	err = g.writeExpressionErrorHandler(indentLevel, attr.Expression)
+	if err != nil {
+		return err
+	}
+
+	// _, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(vn)
+	if _, err = g.w.WriteIndent(indentLevel, "_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString("+vn+"))\n"); err != nil {
+		return err
+	}
+	return g.writeErrorHandler(indentLevel)
+}
+
+func (g *generator) writeExpressionAttributeValueStyle(indentLevel int, attr parser.ExpressionAttribute) (err error) {
+	var r parser.Range
+	vn := g.createVariableName()
+	// var vn string
+	if _, err = g.w.WriteIndent(indentLevel, "var "+vn+" string\n"); err != nil {
+		return err
+	}
+	// vn, templ_7745c5c3_Err = templ.SanitizeStyleAttributeValues(
+	if _, err = g.w.WriteIndent(indentLevel, vn+", templ_7745c5c3_Err = templruntime.SanitizeStyleAttributeValues("); err != nil {
+		return err
+	}
+	// p.Name()
+	if r, err = g.w.Write(attr.Expression.Value); err != nil {
+		return err
+	}
+	g.sourceMap.Add(attr.Expression, r)
+	// )
+	if _, err = g.w.Write(")\n"); err != nil {
+		return err
+	}
+	// Attribute expression error handler.
+	err = g.writeExpressionErrorHandler(indentLevel, attr.Expression)
+	if err != nil {
+		return err
+	}
+
+	// _, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(vn)
+	if _, err = g.w.WriteIndent(indentLevel, "_, templ_7745c5c3_Err = templ_7745c5c3_Buffer.WriteString(templ.EscapeString("+vn+"))\n"); err != nil {
+		return err
+	}
+	return g.writeErrorHandler(indentLevel)
+}
+
 func (g *generator) writeExpressionAttribute(indentLevel int, elementName string, attr parser.ExpressionAttribute) (err error) {
 	attrName := html.EscapeString(attr.Name)
 	// Name
 	if _, err = g.w.WriteStringLiteral(indentLevel, fmt.Sprintf(` %s=`, attrName)); err != nil {
 		return err
 	}
-	// Value.
 	// Open quote.
 	if _, err = g.w.WriteStringLiteral(indentLevel, `\"`); err != nil {
 		return err
 	}
+	// Value.
+	valueWriter := g.writeExpressionAttributeValueDefault
```

Comment:
Wondering if the inderect function call here is worth it for the readability. Probably negligable performance gains though so feel free to ignore!

Reply:
Fair point. I started the other way, but refactored to make it nicer to look at, but lets have the performance instead!